google/cloud/spanner_v1/pool.py

```diff
@@ -423,16 +425,18 @@ def ping(self):
         """
         while True:
             try:
-                ping_after, session = self._sessions.get(block=False)
+                last_used, session = self._sessions.get(block=False)
             except queue.Empty:  # all sessions in use
                 break
-            if ping_after > _NOW():  # oldest session is fresh
+            if last_used + self._delta > _NOW():  # oldest session is fresh
                 # Re-add to queue with existing expiration
-                self._sessions.put((ping_after, session))
+                self._sessions.put((last_used, session))
                 break
-            if not session.exists():  # stale
+            if _NOW() > last_used + _SESSION_EXPIRY_DURATION:  # stale
```

Comment:
Same here. 

What will happen to `session.ping()` if session is deleted?

Reply:
In original testing, pinging a session that had just been deleted would lead to no error. However, adding a delay raises the expected NotFound error. So since the likelihood that it will have just been deleted is low, I'll just rely on the ping to raise the NotFound error.

I'll refactor to rely on ping raising NotFound.